test/crawler/keyword/keywords_test.exs

```diff
@@ -0,0 +1,48 @@
+defmodule Crawler.Keyword.KeywordsTest do
+  use Crawler.DataCase, async: true
+
+  alias Crawler.Keyword.Keywords
+  alias Crawler.Keyword.Schemas.Keyword
+
+  describe("list_keywords/1") do
+    test "given a valid user id, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      keyword = insert(:keyword, user_id: user_id)
+
+      assert Keywords.list_keywords(user_id) == [keyword]
+    end
+  end
+
+  describe("create_keyword/1") do
+    test "given a valid keyword data, returns a saved keyword" do
+      %{id: user_id} = insert(:user)
+      keyword_params = %{name: "keyword", user_id: user_id}
+
+      assert {:ok, %Keyword{} = keyword} = Keywords.create_keyword(keyword_params)
+      assert keyword.name == "keyword"
+    end
+  end
+
+  describe("create_keyword_list/2") do
+    test "given a valid list of keywords, returns a saved keyword list" do
+      %{id: user_id} = insert(:user)
+      Keywords.create_keyword_list(["first", "second"], user_id)
+
+      assert length(Keywords.list_keywords(user_id)) == 2
+    end
+
+    test "given a list with an existing keyword, modifies status of the keyword to new" do
```

Comment:
I'm not sure about this :D, this is not a big deal but for me, it's easy as just to allow the user can re-upload a new keyword instead of updating the status here :D 

And force update the status from `processing` to `new` for a keyword seems not a good idea, as the old keyword is in processing, it might have the Crawler service running and the result is generating, if we force it back to new status, we have to stop the service somewhere later :D 

For me, just go with a simplest solution :D allow the user uploads the same keyword :D

Reply:
@andyduong1920 I understand. Trying to handle several more scenarios seems valuable to me, tbh. And since we're not aligning with any fixed requirement, I will have more chances to practice. Back to `we have to stop the service somewhere later`, I think I can prevent users from uploading new CSV files during processing keywords. What do you think?

Reply:
> Back to we have to stop the service somewhere later, I think I can prevent users from uploading new CSV files during processing keywords. What do you think?

I'm not sure we would like to bring that topic into the IC 😄 your call then, for me, we just need to go with the easier way in the IC, the main purpose of the IC is not `making a better Google crawler service`, it's just learning and practicing in Elixir and Phoenix 👌

Reply:
I think it is a bad idea from the business requirements side.

Search results for different days will be different, and we don't need to lose old search results when we are searching for the same term again (which looks like your idea to do).

Each search result should be kept, and a new DB record should be created if the user is doing a search with the same keyword.

Reply:
Technically, we can store search results in a separate table. Why a keyword has to contain its search result? 
And yeah, this is only for uploading keywords. You haven't known what I will do for the search result.
Btw, I don't want my IC to be delayed because of the better or worse solution architecture here, seems over-skilled for everyone. 🤪 I will remove this logic.

Reply:
> Technically, we can store search results in a separate table.

You can, but you will lose information when the keyword was uploaded and when it was processed when you upload another keyword with the same name until you are going to save this in one more table (which is not done in this PR as well).

Reply:
> over-skilled

I would call it a different term. We shouldn't lose any information if we can avoid that. And simply inserting a new keyword solves this problem and doesn't bring any additional troubles.

Reply:
Yes, you're right. Thanks for the information 🙌

Reply:
We are not looking for the right and wrong person (so, nothing personal here), we are discussing a technical solution that should satisfy business requirements.

Each solution has benefits, disadvantages, and other properties (like implementation time, supporting efforts and etc.). And it is always a trade-off between making everything perfect and making it work a good enough.

As you mentioned before, for this IC it can be simplified, but if you feel strongly about your solution and you have good evidence that it will work better, we can always discuss it further.

Reply:
I think I'm overthinking this. With IC, we can't go far with discussions since reviewers and attendees are limited by the time that could allocate to it, right? 
As I said, `because of the better or worse solution architecture here`, we haven't known which one is better yet, but if we deep dive for clarification, it will take us to a lot of discussions; maybe this PR and the following PRs, and so on.

Reply:
Yeah, the time limit for the developer and reviewers is limited. So, making things simpler will not hurt, I agree.

Reply:
@Nihisil @andyduong1920 Removed unique as fkey in [e125502](https://github.com/vnntsu/elixir-ic/pull/36/commits/e125502b0616069f1785932444a0174d74b4af86). Thanks 🙏